site/docs/red-team/plugins/mcp.md

```diff
@@ -0,0 +1,129 @@
+---
+sidebar_label: MCP Plugin
+---
+
+# MCP Plugin
+
+## Overview
+
+The Model Context Protocol (MCP) Plugin tests whether agentic systems using MCP are vulnerable to function call exploits, system prompt leakage, unauthorized tool discovery, or other MCP-specific attacks.
+
+This plugin maps to multiple threat vectors from the [OWASP Agentic AI Top 10 Threats](https://genai.owasp.org/resource/agentic-ai-threats-and-mitigations/), particularly focusing on Tool Manipulation, System Prompt Extraction, and Privilege Escalation.
+
+## Purpose
+
+MCP enables AI models to use tools, maintain context, and perform complex interactions. However, these capabilities introduce unique vulnerabilities. The MCP Plugin specifically tests for:
+
+1. Exploits that compromise the integrity of function calling mechanisms
+2. Attacks that extract sensitive system information
+3. Techniques that manipulate tool usage in unintended ways
+4. Methods to bypass security controls in MCP implementations
+
+## How it Works
+
+The MCP Plugin:
+
+1. Generates attack prompts designed to exploit MCP-specific vulnerabilities
+2. Analyzes model responses for signs of successful exploitation
+3. Evaluates the robustness of function calling implementations
```

Comment:
What does this mean?

Reply:
I was thinking in terms of parameter sanitation, handling arg types / values, not calling functions with incorrect parameters, etc. I could probably be clearer on that.